packages/opentelemetry-instrumentation-openai/opentelemetry/instrumentation/openai/v1/assistant_wrappers.py

```diff
@@ -62,7 +65,14 @@ def runs_retrieve_wrapper(tracer, wrapped, instance, args, kwargs):
 
     response = wrapped(*args, **kwargs)
 
-    if response.id in runs:
+    if type(response) is LegacyAPIResponse:
+        parsed_response = response.parse()
+    else:
+        parsed_response = response
+
+    assert type(parsed_response) is Run
```

Comment:
@michael-hhai why do you need this?

Reply:
@michael-hhai the problem is that we don't want to throw exceptions in the instrumentations

Reply:
Good to know! We should add `@dont_throw` annotation anyway then